multiversx_sdk_core/proto/transaction_serializer.py

```diff
@@ -23,11 +24,24 @@ class ITransaction(Protocol):
     guardian_signature: bytes
 
 
+class IAddressConverter(Protocol):
+    def __init__(self, hrp: str = DEFAULT_HRP) -> None:
+        ...
+
+    def bech32_to_pubkey(self, value: str) -> bytes:
+        ...
+
+    def pubkey_to_bech32(self, pubkey: bytes) -> str:
+        ...
+
+
 class ProtoSerializer:
+    def __init__(self, converter: IAddressConverter) -> None:
+        self.converter = converter
```

Comment:
Can be named `address_converter` (more explicit), in case we'll have some other converters here (assumption, speculation).

Reply:
Renamed to `address_converter`